ITISFoundation/osparc-simcore
# 📌 Maintenance/webserver requirements

## What do these changes do?

- fixes ``web/server/requirements`` : removes ``packages.*`` files and old constraints.
- Add extra dependency tools (``pipreqs`` and ``pipreqtree`` to assist resolving conflicts and analyzing dependencies)

## Related issue/s



## How to test

Check that the requirements ``_base.txt`` are the libraries that actually get installed in production:
```cmd
$ make devenv
$ source .venv/bin/activate
$ cd services/web/server

$ make install-prod

$ make pip-freeze

$  diff -i freeze-base.ignore.txt freeze-now.ignore.txt
32c32
< importlib-resources==5.4.0 ; python_version < "3.9"
---
> importlib-resources==5.4.0
50a51,52
> pep517==0.12.0
> pip-tools==6.4.0
62a65,69
> simcore-models-library @ file:///home/crespo/devp/osparc-simcore/packages/models-library
> simcore-postgres-database @ file:///home/crespo/devp/osparc-simcore/packages/postgres-database
> simcore-service-library @ file:///home/crespo/devp/osparc-simcore/packages/service-library
> simcore-service-webserver @ file:///home/crespo/devp/osparc-simcore/services/web/server
> simcore-settings-library @ file:///home/crespo/devp/osparc-simcore/packages/settings-library
66a74
> tomli==1.2.3
76d83
< 
```
Notice that the differences are tools (i.e. pip-tools etc) and osparc-simcore packages (which are not in ``_base.py``)


## Checklist
